tests/api2/test_007_early_settings.py

```diff
@@ -14,13 +14,40 @@ def test_sysctl_arc_max_is_set():
     assert call('filesystem.stat', DEFAULT_ARC_MAX_FILE)['size']
 
 
-def test_database_mode():
-    """Test the mode of the database file."""
-    db_name = "/data/freenas-v1.db"
-    db_mode = 0o600
-    mode = call('filesystem.stat', db_name)['mode']
-    assert mode & 0o777 == db_mode, mode
-    if ha:
-        call('failover.sync_to_peer')
-        mode = call('failover.call_remote', 'filesystem.stat', [db_name])['mode']
-        assert mode & 0o777 == db_mode, mode
+def test_data_dir_perms():
+    default_dir_mode = 0o700
+    default_file_mode = 0o600
+    data_subsystems_mode = 0o755
+
+    def check_permissions(directory):
+        contents = call('filesystem.listdir', directory)
+        for i in contents:
+            i_path = i['path']
+            i_mode = call('filesystem.stat', i_path)['mode']
+            if i['type'] == 'DIRECTORY':
```

Comment:
If we have mode in listdir output then you can simply import `stat` module and call `stat.S_ISDIR(i['mode'])`